Add hal.h from the prima driver

kvalo opened the pull request:
This pull requests adds hal.h from the prima driver. The benefit from this is that we use the same terms as the prima driver.

I did a lot of cleanup but there's still few pendings pending, for example endian support, few typedefs and structure field names. I'll clean those later.

Commit:
Fix status size in START_RSP message

In START_RSP message status filed is 2 bytes but not 4 as in
all other responses. Because of that check for status was
always failing all the time. As a fix check for success only
2 bytes.

Signed-off-by: Eugene Krasnikov <eugene.krasnikov@sonymobile.com>

kvalo committed:
wcn36xx: import hal.h from the prima driver

URL:

https://www.codeaurora.org/gitweb/external/wlan/?p=prima.git;a=blob_plain;f=riva/inc/wlan_hal_msg.h;h=f29aaf66e59c566cde4bd99bc2b00024641aac21;hb=HEAD

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: make hal.h compile

So that we can use it with wcn36xx.

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
add hw.h

From URL:

https://www.codeaurora.org/gitweb/external/wlan/?p=prima.git;a=blob_plain;f=riva/inc/wlan_qct_dev_defs.h;h=b9577e6bd173b3e44d83bfc40d852b381585c6ca;hb=HEAD

hal.h needs these

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: remove ugly typedefs and PACKED macros from structs

Follow Linux style and use real structs and __packed.

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: remove all packed macros

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: run Lindent

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: comment cleanup

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: remove useless ifdefs

Leave the ones which change structure content, these need to be solved somehow differently.

Also fix few compiler errors which happened due to #ifdef removal.

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: use WCN36XX_HAL prefix for command ids

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: reindent PARAM_BEACON_UPDATE_MASK

Lindent reformatted it in a really ugly way.

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: get rid of tSirMacAddr

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: get rid of most typedefs

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

KrasnikovEugene commented:
enums like eSIR_WEP_STATIC  must be renamed

KrasnikovEugene commented:
enums like eSTA_11ac must be removed since HW is not supporting 11ac

KrasnikovEugene commented:
suggest also to change variable name to something more linux friendli e.g.  ucMaxBssids change to max_bssids

kvalo committed:
hal: remove hungarian notation from enums

The 'e' prefix is useless.

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

KrasnikovEugene commented:
Also suggest to change PE_SUMMARY_STATS_INFO to something else because PE is a similar to SIR(one of the layers in prima driver) and not valid for wcn36xx any more

kvalo commented:
In most cases we should use WCN36XX_HAL where it’s not excessive

KrasnikovEugene notifications@github.com writes:

> Also suggest to change PE_SUMMARY_STATS_INFO to something else because PE is a
> similar to SIR(one of the layers in prima driver) and not valid for wcn36xx any
> more

Maybe I should use WCN36XX_HAL_ and wcn36xx_hal_ prefixes everywhere?

## 

Kalle Valo

kvalo committed:
delete hw.h

We actually just need just HAL_NUM_BSSID from hw.h. So copy that to hal.h and delete hw.h entirely.

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

KrasnikovEugene commented on hw.h:
```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (c) 2012, The Linux Foundation. All rights reserved.
+ *
+ * Previously licensed under the ISC license by Qualcomm Atheros, Inc.
```
copyrights twice. probably to excessive;)

kvalo replied:
KrasnikovEugene notifications@github.com writes:

> In hw.h:
> 
> > - \* above copyright notice and this permission notice appear in all
> > - \* copies.
> > - *
> > - \* THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
> > - \* WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
> > - \* WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
> > - \* AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
> > - \* DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR
> > - \* PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
> > - \* TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
> > - \* PERFORMANCE OF THIS SOFTWARE.
> > - _/
> >   +/_
> > - \* Copyright (c) 2012, The Linux Foundation. All rights reserved.
> > - *
> > - \* Previously licensed under the ISC license by Qualcomm Atheros, Inc.
> 
> copyrights twice. probably to excessive;)

I have already removed hw.h from my branch already, we just needed one
define from hw.h.

## 

Kalle Valo

KrasnikovEugene commented:
I suggest to remove this.

#define IS_MCC_SUPPORTED_BY_HOST (!!(halMsg_GetHostWlanFeatCaps(MCC)))
#define IS_SLM_SESSIONIZATION_SUPPORTED_BY_HOST (!!(halMsg_GetHostWlanFeatCaps(SLM_SESSIONIZATION)))
#define IS_FEATURE_SUPPORTED_BY_HOST(featEnumValue) (!!halMsg_GetHostWlanFeatCaps(featEnumValue))
#define IS_WLANACTIVE_OFFLOAD_SUPPORTED_BY_HOST (!!(halMsg_GetHostWlanFeatCaps(WLANACTIVE_OFFLOAD)))

kvalo committed:
hal: clean up structure field names

Simplify them and get rid of camel case style.

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: remove unused feature macros

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: remove tHalIpv4Addr typedef

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: whitespace or comment cleanup

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: rename sir_ prefix to wcn36xx_hal_

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: remove all typedefs

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: hal_msg_header -> wcn36xx_hal_msg_header

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: rename hal_ prefix to wcn36xx_hal_

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: kill WCN36XX_HAL_MAC_ADDR_LEN

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

kvalo committed:
hal: name unions with u

Signed-off-by: Kalle Valo <kvalo@qca.qualcomm.com>

KrasnikovEugene commented:
Ok, there is so many changes already so we have to merge this and start to use it. Let's just file issues when we found something.
